runtime/bundles/org.eclipse.core.jobs/src/org/eclipse/core/internal/jobs/Semaphore.java

```diff
@@ -31,6 +31,12 @@ public Semaphore(Runnable runnable) {
 	 * and false otherwise.
 	 */
 	public synchronized boolean acquire(long delay) throws InterruptedException {
+		if (notifications == 0) {
```

Comment:
This does not make sense, that way the process will **never** wait... because there are only two cases:

1. notifications = 0 (semaphore is blocked) --> throws exception
2. notifications = 1 and will return immediately